src/components/action-button/action-button.spec.tsx

```diff
@@ -0,0 +1,107 @@
+import { screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import { renderWithTheme } from "../../utils/testing-utils/render-with-theme";
+import { ActionButton } from "./action-button";
+import { ActionButtonVariant } from "../../types";
+
+describe("Action Button", () => {
+  const onClick = jest.fn();
+
+  it("Should render primary button by default", () => {
+    const { container } = renderWithTheme(
+      <ActionButton onClick={onClick}
+      >
+        Action Button
+      </ActionButton>
+    );
+    expect(container).toBeInTheDocument();
+
+    const bgColor = getComputedStyle(screen.getByText("Action Button") as Element).getPropertyValue(
+      "background-color"
+    );
+
+    expect(bgColor).toBe("rgb(96, 93, 236)");
+
+    userEvent.click(screen.getByText("Action Button"));
+    expect(onClick).toHaveBeenCalled();
+  });
+
+  it("Should render primary button by props", () => {
+    const { container } = renderWithTheme(
+      <ActionButton
+        variant={ActionButtonVariant.primary}
+        onClick={onClick}
+      >
+        Action Button
+      </ActionButton>
+    );
+    expect(container).toBeInTheDocument();
+
+    const button = screen.getByText("Action Button")
+    const bgColor = getComputedStyle(button as Element).getPropertyValue(
+      "background-color"
+    );
+
+    expect(bgColor).toBe("rgb(96, 93, 236)");
+    userEvent.click(button);
+    expect(onClick).toHaveBeenCalled();
+  });
+
+
+  it("Should render secondary button", () => {
+    const { container } = renderWithTheme(
+      <ActionButton
+        variant={ActionButtonVariant.secondary}
+        onClick={onClick}
+      >
+        Action Button
+      </ActionButton>
+    );
+    expect(container).toBeInTheDocument();
+
+    const button = screen.getByText("Action Button")
+    const bgColor = getComputedStyle(button as Element).getPropertyValue(
+      "background-color"
+    );
+
+    expect(bgColor).toBe("transparent");
+    userEvent.click(button);
+    expect(onClick).toHaveBeenCalled();
+  });
+
+  it("Should render cancel button", () => {
+    const { container } = renderWithTheme(
+      <ActionButton
+        variant={ActionButtonVariant.cancel}
+        onClick={onClick}
+      >
+        Action Button
+      </ActionButton>
+    );
+    expect(container).toBeInTheDocument();
+
+    const button = screen.getByText("Action Button")
+    const bgColor = getComputedStyle(button as Element).getPropertyValue(
+      "background-color"
+    );
+
+    expect(bgColor).toBe("transparent");
```

Comment:
secondary and cancel - same check. Add checks to clarify difference between those types

Reply:
Added check for border color as well, they have differecies.